src/dmd/mars.d

```diff
@@ -1361,7 +1361,37 @@ void addDefaultVersionIdentifiers()
     static if (TARGET.Windows)
     {
         if (global.params.mscoff)
+        {
             VersionCondition.addPredefinedGlobalIdent("CRuntime_Microsoft");
+            if (!global.params.mscrtlib || !global.params.mscrtlib[0])
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_None");
+            }
+            else if (!strcmp(global.params.mscrtlib, "libcmt"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Static");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Release");
+            }
+            else if (!strcmp(global.params.mscrtlib, "libcmtd"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Static");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Debug");
+            }
+            else if (!strcmp(global.params.mscrtlib, "msvcrt"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Dynamic");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Release");
+            }
+            else if (!strcmp(global.params.mscrtlib, "msvcrtd"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Dynamic");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Debug");
+            }
+            else
+            {
+                assert(0, "unexpected mscrtlib, how did the command line accept this?");
```

Comment:
Please, no snark in the code.

Reply:
That's not even remotely snark... that's a genuine question.
In the event of that assert, the question is, how did the commend line allow it? (ie, that is the problem source)

Reply:
I'd prefer "the command line should have rejected it"